File: app/src/main/java/com/ssz/framejava/ui/home/HomePresenter.java
package com.ssz.framejava.ui.home;

import javax.inject.Inject;

/**
 * @author : zsp
 * time : 2019 12 2019/12/23 15:20
 */
public class HomePresenter implements IHomeContract.IPresenter {

    private IHomeContract.IView mView;

    @Inject HomePresenter(IHomeContract.IView view){
        this.mView = view;
    }

    @Override
    public void detach() {
        this.mView = null;
    }
}

File: app/src/main/java/com/ssz/studydemo/ui/HomePresenter.kt
package com.ssz.studydemo.ui

import com.ssz.studydemo.ui.home.IHomeContract

/**
 * @author : zsp
 * time : 2019 09 2019/9/17 15:11
 */
class HomePresenter(val view : IHomeContract.IHomeView) : IHomeContract.IHomePresenter {

    init {
        view.mPresenter = this
    }
}